metrics/query.response-time/response_time.go

```diff
@@ -179,15 +209,60 @@ func (c *ResponseTime) Collect(ctx context.Context, levelName string) ([]blip.Me
 	}
 
 	if c.atLevel[levelName].truncate {
-		_, err := c.db.Exec(TRUNCATE_QUERY)
+		trCtx, cancelFn := context.WithTimeout(ctx, c.atLevel[levelName].truncateTimeout)
+		defer cancelFn()
+		_, err := c.db.ExecContext(trCtx, TRUNCATE_QUERY)
 		if err != nil {
-			return nil, err
+			return c.collectTruncateError(err, levelName, metrics)
+		} else if c.atLevel[levelName].hadTruncateError {
+			// Upon success we can start emitting metrics normally again but we need to
+			// wait for the next collect for the truncate to have taken effect.
+			// Emit whatever metrics would be emitted based on the error policy.
+			metrics, err = c.collectTruncateError(fmt.Errorf("Truncate successful, waiting for next metric collection."), levelName, metrics)
 		}
+		c.atLevel[levelName].hadTruncateError = false
```

Comment:
I know what this stuff is doing, but it's a little hard to follow. It's something like:
1. error
2. success (don't report)
3. success (ok now start reporting)
I'd see if there's a way to make it more clear, either in code or with comments.